compiler/p/codegen/PPCBinaryEncoding.cpp

```diff
@@ -1019,6 +1019,20 @@ static void fillFieldR(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
    cursor[0] |= (val << 20);
    }
 
+/**
+ * Fills in the 8-bit IMM8 field of a binary-encoded instruction with the provided immediate value:
+ *
+ * +----------------------+---------------------------------------+
+ * |                      | IMM8            |                     |
+ * | 0                    | 13              | 21                  |
+ * +----------------------+---------------------------------------+
+ */
+static void fillFieldIMM8(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, isValidInSignExtendedField(val, 0xffu), "0x%x is out-of-range for IMM8(8) field", val);
```

Comment:
I think this should be changed to also allow unsigned 8 bit values as well as signed 8 bit values. This field is a bit unusual in how it accepts both signed or unsigned 8 bit immediate values since there's no need to zero extend or sign extend.

Reply:
I agree with @IBMJimmyk . Although we might not generate such code yet, the instruction allows the immediate value to be unsigned byte as well. But I don't see a method for checking for unsigned byte.

Reply:
I think I accidentally removed my comment. I agree with @IBMJimmyk : I think we should allow unsigned byte values too. Although I am not sure we have a method for that.